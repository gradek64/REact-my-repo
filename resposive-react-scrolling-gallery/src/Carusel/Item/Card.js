import React from 'react';
import PropTypes from 'prop-types';

const Card = ({
  index, picture, city, address, bedrooms, bathrooms, carSpaces,
}) => (
  <li id={`card-${index}`} className="card">
    <img src={picture} alt={city} />
    <div className="details">
      <span className="index">{index + 1}</span>
      <p className="location">
        {city}
        <br />
        {address}
      </p>
      <ul className="features">
        <li className="icon-bed">
          {bedrooms}
          {' '}
          <span>bedrooms</span>
        </li>
        <li className="icon-bath">
          {bathrooms}
          {' '}
          <span>bathrooms</span>
        </li>
        <li className="icon-car">
          {carSpaces}
          {' '}
          <span>parking spots</span>
        </li>
      </ul>
    </div>
  </li>
);

Card.propTypes = {
  index: PropTypes.number.isRequired,
  picture: PropTypes.string.isRequired,
  city: PropTypes.string.isRequired,
  address: PropTypes.string.isRequired,
  bedrooms: PropTypes.number.isRequired,
  bathrooms: PropTypes.number.isRequired,
  carSpaces: PropTypes.number.isRequired,
};

export default Card;
