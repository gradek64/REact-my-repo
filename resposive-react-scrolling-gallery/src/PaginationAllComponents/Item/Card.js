import React from 'react';
import PropTypes from 'prop-types';

const Card = ({
  index, picture, city, address, bedrooms, bathrooms, carSpaces,
}) => (
  <li id={`card-${index}`} className="card">
    <img src={picture} alt={city} />
  </li>
);

Card.propTypes = {
  index: PropTypes.number.isRequired,
  picture: PropTypes.string.isRequired,
  city: PropTypes.string.isRequired,
  address: PropTypes.string.isRequired,
  bedrooms: PropTypes.number.isRequired,
  bathrooms: PropTypes.number.isRequired,
  carSpaces: PropTypes.number.isRequired,
};

export default Card;
