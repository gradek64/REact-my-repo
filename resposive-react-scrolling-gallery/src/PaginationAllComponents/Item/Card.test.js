import React from 'react';
import { shallow } from 'enzyme';
import Card from './Card';

const props = {
  index: 0,
  picture: 'images/tn_property01.jpg',
  city: 'Singer',
  address: '914 Argyle Road',
  bedrooms: 2,
  bathrooms: 2,
  carSpaces: 2,
};

test('renders correctly', () => {
  const wrapper = shallow(<Card {...props} />);
  expect(wrapper).toMatchSnapshot();
});
