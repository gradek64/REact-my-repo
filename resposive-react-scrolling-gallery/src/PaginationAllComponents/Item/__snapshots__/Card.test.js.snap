// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<li
  className="card"
  id="card-0"
>
  <img
    alt="Singer"
    src="images/tn_property01.jpg"
  />
</li>
`;
